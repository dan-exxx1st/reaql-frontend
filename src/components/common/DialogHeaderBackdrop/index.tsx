import React from 'react';
import {
    StyledDialogHeaderBackdropWrapper,
    StyledDialogHeaderBackdropIcon,
    StyledDialogHeaderBackdropText,
    StyledDialogHeaderBackdropListWrapper,
    StyledDialogHeaderBackdropTriangle,
} from './style';

import PlusIcon from 'assets/images/icons/BluePlus.svg';
import SearchIcon from 'assets/images/icons/BlueSearch.svg';
import TrashIcon from 'assets/images/icons/BlueTrash.svg';

const DialogHeaderBackdrop = () => (
    <StyledDialogHeaderBackdropWrapper>
        <StyledDialogHeaderBackdropTriangle />
        <StyledDialogHeaderBackdropListWrapper>
            <StyledDialogHeaderBackdropIcon src={PlusIcon} />
            <StyledDialogHeaderBackdropText variant="body2">
                Add members
            </StyledDialogHeaderBackdropText>
        </StyledDialogHeaderBackdropListWrapper>

        <StyledDialogHeaderBackdropListWrapper>
            <StyledDialogHeaderBackdropIcon src={SearchIcon} />
            <StyledDialogHeaderBackdropText variant="body2">
                Search message history
            </StyledDialogHeaderBackdropText>
        </StyledDialogHeaderBackdropListWrapper>

        <StyledDialogHeaderBackdropListWrapper>
            <StyledDialogHeaderBackdropIcon src={TrashIcon} />
            <StyledDialogHeaderBackdropText variant="body2">
                Clear message history
            </StyledDialogHeaderBackdropText>
        </StyledDialogHeaderBackdropListWrapper>
    </StyledDialogHeaderBackdropWrapper>
);

export default DialogHeaderBackdrop;
