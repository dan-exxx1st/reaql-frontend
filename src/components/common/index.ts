export { default as DialogHeader } from './DialogHeader';
export { default as UserHeader } from './UserHeader';

export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';

export { default as MessageSearch } from './MessageSearch';
export { default as DialogTextField } from './DialogTextField';

export { default as ContactItem } from './ContactItem';
export { default as MessageItem } from './MessageItem';

export { default as ContactList } from './ContactList';
export { default as MessageList } from './MessageList';
