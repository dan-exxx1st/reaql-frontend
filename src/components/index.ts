export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
