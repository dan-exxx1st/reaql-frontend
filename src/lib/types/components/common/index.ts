import Theme from 'helpers/styled';
import { Dialog, Maybe, Message, Message_Statuses } from 'lib/graphql/types';
import { Dispatch, SetStateAction } from 'react';

export interface IUserHeaderProps {
    className?: string;
    setSearchUserOpened?: (val: boolean) => void;
}

export interface IMessageSearchProps {
    onClick?: () => void;
    value?: string;
    messageFilter?: (value: string) => void | Dispatch<SetStateAction<string>>;
}

export interface IMessageItemProps extends Message {
    className?: string;
}

export interface IDialogsItemProps {
    id: string;
    name: string;
    surname: string;
    avatar: string;
    unreadMessages: number | null | undefined;
    lastMessageStatus: Message_Statuses | null | undefined;
    lastMessage: string | null | undefined;
    lastMessageDate: string | null | undefined;
}

export interface IMessageListProps {
    data?: {
        messages: Message[];
    };
    loading?: boolean;
    subscribeToNewMessages?: () => void;
    className?: string;
    filterValue?: string;
}

export interface IDialogListProps {
    dialogFilter: string;
    dialogs?: Maybe<Dialog[]>;
    className?: string;
}

export interface IDialogTextFieldProps {
    className?: string;
}

export interface IDialogHeaderProps {
    className?: string;
    messageFilter?: Dispatch<SetStateAction<string>>;
    filterValue?: string;
    name: string;
    onlineStatus?: string | null;
    avatar?: string;
    loading?: boolean;
}

export interface ISignUpFormProps {
    email: string;
    name: string;
    surname: string;
    password: string;
    changeFormValues: (field: string, value: string) => void;
    handleSubmit: () => void;
    theme?: typeof Theme;
}

export interface ISignInFormProps {
    _handleSignIn: (payload: {
        email: string;
        password: string;
        rememberMe: boolean;
    }) => void;
}

export interface ISideBarProps {
    data?: {
        dialogs: Dialog[];
    };
    loading: boolean;
    subscribeToNewDialogs?: () => void;
    setSearchUserOpened: (val: boolean) => void;
}
