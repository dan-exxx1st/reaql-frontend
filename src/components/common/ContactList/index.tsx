import React, { FC } from 'react';

import { IContactListProps } from 'lib/types/components/common';

import { StyledContactListWrapper, StyledContactListItem } from './style';

const ContactList: FC<IContactListProps> = ({ contacts, ...otherProps }) => {
    return (
        <StyledContactListWrapper {...otherProps}>
            {contacts &&
                contacts.map((contact) => (
                    <StyledContactListItem key={contact.id} {...contact} />
                ))}
        </StyledContactListWrapper>
    );
};

export default ContactList;
