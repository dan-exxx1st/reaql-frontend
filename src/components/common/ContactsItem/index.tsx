import React, { FC, useEffect, useState } from 'react';
import StyledDialogListItem, {
    StyledContactsItemAvatarWrapper,
    StyledContactsItemCircle,
    StyledContactsItemBottomLine,
} from './style';
import { Wrapper, Avatar, Typography, CheckMark } from 'components/UI';
import { IContactsItemProps } from 'lib/types/components/common';

const ContactItem: FC<IContactsItemProps> = ({
    avatar_src,
    name,
    last_message,
    last_message_date,
    unread_message_count,
    sended_message_status,
}) => {
    const [isDoubleCheckMark, setIsDoubleCheckMark] = useState(false);
    const [isActiveCheckMark, setIsActiveCheckMark] = useState(false);
    const [undreadMessageCountText, setUnreadMessageCountText] = useState('');

    useEffect(() => {
        switch (sended_message_status) {
            case 'sended': {
                setIsDoubleCheckMark(false);
                setIsActiveCheckMark(false);
                break;
            }

            case 'received': {
                setIsDoubleCheckMark(true);
                break;
            }

            case 'readed': {
                setIsActiveCheckMark(true);
            }
        }
        if (unread_message_count) {
            if (unread_message_count > 0 && unread_message_count < 10) {
                setUnreadMessageCountText(unread_message_count.toString());
            } else {
                setUnreadMessageCountText('9+');
            }
        }
    }, []);

    return (
        <StyledDialogListItem flexWrap="wrap">
            <Wrapper height="40px" width="100%" justifyContent="space-between">
                <Wrapper height="40px">
                    <StyledContactsItemAvatarWrapper>
                        <Avatar src={avatar_src} alt="Avatar image" />
                    </StyledContactsItemAvatarWrapper>
                    <Wrapper
                        height="40px"
                        flexDirection="column"
                        justifyContent="space-between"
                    >
                        <Typography variant="body1" color="dc">
                            {name}
                        </Typography>
                        <Typography variant="caption1" color="dsc">
                            {last_message}
                        </Typography>
                    </Wrapper>
                </Wrapper>
                <Wrapper
                    flexDirection="column"
                    height="40px"
                    justifyContent="space-between"
                    alignItems="flex-end"
                >
                    <Typography variant="caption1" color="dgc">
                        {last_message_date?.toString()}
                    </Typography>

                    <Wrapper>
                        {undreadMessageCountText ? (
                            <Wrapper>
                                <StyledContactsItemCircle
                                    color="primary"
                                    text={undreadMessageCountText}
                                />
                            </Wrapper>
                        ) : (
                            <Wrapper>
                                <CheckMark
                                    isDouble={isDoubleCheckMark}
                                    isChecked={isActiveCheckMark}
                                />
                            </Wrapper>
                        )}
                    </Wrapper>
                </Wrapper>
            </Wrapper>

            <StyledContactsItemBottomLine />
        </StyledDialogListItem>
    );
};

export default ContactItem;
