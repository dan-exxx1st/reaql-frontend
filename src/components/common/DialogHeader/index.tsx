import React, { FC, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { decode } from 'querystring';

import { Typography, Wrapper } from 'components/UI';
import { MessageSearch } from '..';

import { StyledAvatar, StyledCircle, StyledDialogHeader } from './style';

import { IDialogHeaderProps } from 'lib/types/components/common';
import { Query } from 'lib/graphql/types';
import { GET_DIALOG } from 'lib/graphql/queries/dialog';

const DialogHeader: FC<IDialogHeaderProps> = ({ className }) => {
    const [isShowMessageSearch, setIsShowMessageSearch] = useState(false);

    const history = useHistory();
    const dialogId = decode(history.location.search)['?dialog'];

    const { data } = useQuery<Query>(GET_DIALOG, {
        variables: { dialogId },
    });

    const _handleToggleMessageSearch = () => {
        setIsShowMessageSearch(!isShowMessageSearch);
    };

    if (data && data.dialog) {
        const {
            dialog: { users, group },
        } = data;
        const name =
            !group && users[0]
                ? `${users[0].name} ${users[0].surname}`
                : 'group!!!'; //!SERVER
        const avatar = !group && users[0] ? users[0].avatar : ''; //!SERVER

        return (
            <StyledDialogHeader
                className={className}
                justifyContent="space-between"
            >
                {isShowMessageSearch ? (
                    <MessageSearch onClick={_handleToggleMessageSearch} />
                ) : (
                    <>
                        <Wrapper>
                            <StyledAvatar src={avatar} />
                            <Wrapper
                                flexDirection="column"
                                justifyContent="space-between"
                                height="40px"
                            >
                                <Typography variant="body1" color="dc">
                                    {name}
                                </Typography>
                                <Typography variant="caption1" color="dgc">
                                    Today at 3:30 pm
                                </Typography>
                            </Wrapper>
                        </Wrapper>
                        <Wrapper>
                            <StyledCircle
                                color="secondary"
                                iconName="search"
                                onClick={_handleToggleMessageSearch}
                            />
                            <StyledCircle
                                color="secondary"
                                iconName="threeDotsGrey"
                            />
                        </Wrapper>
                    </>
                )}
            </StyledDialogHeader>
        );
    }

    return <div>Loading...</div>;
};

export default DialogHeader;
