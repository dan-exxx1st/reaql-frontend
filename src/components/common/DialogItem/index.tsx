import React, { FC, useEffect, useState } from 'react';
import StyledDialogListItem, {
    StyledDialogItemAvatarWrapper,
    StyledDialogItemCircle,
    StyledDialogItemRightWrapper,
} from './style';
import { Wrapper, Avatar, Typography, CheckMark } from 'components/UI';
import { IDialogsItemProps } from 'lib/types/components/common';
import { Message_Statuses } from 'lib/graphql/types';
import { useHistory } from 'react-router-dom';

const DialogItem: FC<IDialogsItemProps> = (props) => {
    const {
        id,
        lastMessageStatus,
        unreadMessages,
        avatar,
        name,
        surname,
        lastMessage,
        lastMessageDate,
    } = props;

    const history = useHistory();

    const [isDoubleCheckMark, setIsDoubleCheckMark] = useState(false);
    const [isActiveCheckMark, setIsActiveCheckMark] = useState(false);
    const [undreadMessageCountText, setUnreadMessageCountText] = useState('');

    const _handleClick = () => {
        history.push(`?dialog=${id}`);
    };

    useEffect(() => {
        if (lastMessageStatus) {
            switch (lastMessageStatus) {
                case Message_Statuses.Sended: {
                    setIsDoubleCheckMark(false);
                    setIsActiveCheckMark(false);
                    break;
                }

                case Message_Statuses.Recived: {
                    setIsDoubleCheckMark(true);
                    setIsActiveCheckMark(false);
                    break;
                }

                case Message_Statuses.Readed: {
                    setIsDoubleCheckMark(true);
                    setIsActiveCheckMark(true);
                }
            }
        }
        if (unreadMessages) {
            if (unreadMessages > 0 && unreadMessages < 10) {
                setUnreadMessageCountText(unreadMessages.toString());
            } else {
                setUnreadMessageCountText('9+');
            }
        }
    }, []);

    return (
        <StyledDialogListItem flexWrap="wrap" onClick={_handleClick}>
            <Wrapper height="40px" width="100%" justifyContent="space-between">
                <Wrapper height="40px">
                    <StyledDialogItemAvatarWrapper>
                        <Avatar src={avatar} alt="Avatar image" />
                    </StyledDialogItemAvatarWrapper>
                    <Wrapper
                        height="40px"
                        flexDirection="column"
                        justifyContent="space-between"
                    >
                        <Typography variant="body1" color="dc">
                            {`${name} ${surname}`}
                        </Typography>
                        <Typography variant="caption1" color="dsc">
                            {lastMessage}
                        </Typography>
                    </Wrapper>
                </Wrapper>
                <StyledDialogItemRightWrapper
                    flexDirection="column"
                    height="40px"
                    justifyContent="space-between"
                    alignItems="flex-end"
                >
                    <Typography variant="caption1" color="dgc">
                        {lastMessageDate?.toString()}
                    </Typography>

                    <Wrapper>
                        {undreadMessageCountText ? (
                            <Wrapper>
                                <StyledDialogItemCircle
                                    color="primary"
                                    text={undreadMessageCountText}
                                />
                            </Wrapper>
                        ) : (
                            <Wrapper>
                                <CheckMark
                                    isDouble={isDoubleCheckMark}
                                    isChecked={isActiveCheckMark}
                                />
                            </Wrapper>
                        )}
                    </Wrapper>
                </StyledDialogItemRightWrapper>
            </Wrapper>
        </StyledDialogListItem>
    );
};

export default DialogItem;
