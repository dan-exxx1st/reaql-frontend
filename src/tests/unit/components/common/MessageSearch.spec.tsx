import React from 'react';
import { MessageSearch } from 'components/common';
import {
    mountWithApolloAndStyled,
    shallowWithApolloAndStyled,
} from 'tests/helpers/withApollo';

describe('<MessageSearch />', () => {
    describe('Snapshots', () => {
        it('Should render correctly', () => {
            const wrapper = mountWithApolloAndStyled(<MessageSearch />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    it('Should use onClick event', () => {
        const onClickEvent = jest.fn();
        const wrapper = shallowWithApolloAndStyled(
            <MessageSearch onClick={onClickEvent} />
        );

        wrapper
            .dive()
            .dive()
            .dive()
            .dive()
            .dive()
            .find('Styled(Styled(Button))')
            .simulate('click');

        expect(onClickEvent).toHaveBeenCalledTimes(1);
    });
});
