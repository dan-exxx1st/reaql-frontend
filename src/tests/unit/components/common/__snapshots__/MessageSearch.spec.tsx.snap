// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageSearch /> Snapshots Should render correctly 1`] = `
.c6 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  margin: 0;
}

.c5 {
  border-radius: 5px;
  outline: none;
  border: none;
  font-family: 'Roboto',sans-serif;
  padding: 7px 26px;
  cursor: pointer;
  max-height: 50px;
  background: #E6E6E6;
  color: #A5A5A5;
}

.c5:hover {
  box-shadow: 1px 1px 10px rgb(10,10,10,0.1);
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c2 {
  position: static;
}

.c4 {
  word-break: break-all;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Roboto',sans-serif;
  padding: 10px;
  border: 1px solid #E6E6E6;
  border-radius: 5px;
  outline: none;
  color: #555555;
  cursor: text;
  font-size: 14px;
  font-weight: normal;
  line-height: 16px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  width: 190px;
  height: auto;
  padding-right: 0;
}

.c4 ~ img {
  width: 20px;
  height: 20px;
  position: absolute;
  top: 50%;
  right: 10px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c1 {
  width: 100%;
  max-width: 976px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  margin-right: 20px;
  width: calc(100% - 120px);
}

.c3 > input {
  width: calc(100% - 20px);
  padding: 7px 10px;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  >
    <input
      className="c4"
      color="secondary"
      onChange={[Function]}
      placeholder="Search message history"
    />
  </div>
  <button
    className="c5"
  >
    <p
      className="c6"
    >
      Cancel
    </p>
  </button>
</div>
`;
